docs/design/backup_and_restore.md

```diff
@@ -0,0 +1,202 @@
+
+#### Toggle Backup Running
+
+In the `Toggle Backup Running` stage, we will start or stop the backup, if necessary. If the desired state is `stopped`, and there is a backup running on the configured tag, this will run the `fdbbackup abort` command. If the desired state is `running` or `paused`, and there is no backup running on the configured tag, this will run the `fdbbackup start` command.
+
+In order to support restarting backups when configuration changes, the operator will automatically include a timestamp when it starts a backup. The format of the timestamp will be configurable, and configuring it to an empty string will omit the timestamp. When a change is detected that requires restarting a backup, we will stop the backup and start a new one. We will continue to track both backups in the status of the resource until the old backup is fully deleted.
+
+Doing the initial snapshot for a backup will require more bandwidth than subsequent snapshots, because the initial snapshot tries to complete as fast as it can while subsequent snapshots complete more slowly based on the snapshot interval. If we update all backups at once, it would cause every backup to stop and then start a new backup. This creates a gap in restorability until the new backup completes its initial snapshot. If all of the backups are doing this at once, it could lead to rate limiting on the destination that causes all of the backups to slow down, creating a longer aggregate gap in restorability. To prevent this, we will have a cross-backup throttle to limit how many backups can be in the initial snapshot phase at a time. The default limit will be 1. If this limit has already been reached, then the operator will not start or restart any backups, but it will be able to stop a backup if the backup has a desired state of `stopped`. This limit will only apply within a single KC.
+
+Changing the following fields will require restarting a backup:
+
+* Backup destination
+* Tag
+* Key Range
+
+#### Modify Backup
+
+In the `Modify Backup` stage, we will update parameters for the backup. If the backup parameters do not match the configured set, this will run the `fdbbackup modify` command.
+
+Changing the following fields will require running the modify command:
+
+* Backup URL parameters
+* Snapshot time
+
+#### Toggle Backup Paused
+
+In the `Toggle Backup Paused` stage, we will pause or resume the backup, if necessary. If the desired state is `paused` and the backup is not paused, this will run the `fdbbackup pause` command. If the desired state is `running` and the backup is paused, this will run the `fdbbackup resume` command.
```

Comment:
We need this step to ensure we did the initial backup once? Otherwise I'm not sure if we really need this step of if we could merge it with `Toggle Backup Running`.

Reply:
You're right. These two actions could probably be merged into a single action.